src/main/scala/io/iohk/ethereum/testmode/TestModeWorldStateProxy.scala

```diff
@@ -0,0 +1,136 @@
+package io.iohk.ethereum.testmode
+
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage.{EvmCodeStorage, MptStorage}
+import io.iohk.ethereum.db.storage.EvmCodeStorage.Code
+import io.iohk.ethereum.domain.Account.accountSerializer
+import io.iohk.ethereum.domain.{Account, Address, UInt256}
+import io.iohk.ethereum.ledger.{InMemorySimpleMapProxy, InMemoryWorldStateProxy, InMemoryWorldStateProxyStorage}
+import io.iohk.ethereum.mpt.MerklePatriciaTrie
+
+/** This is a wrapper around InMemoryWorldStateProxy.
+  * Its only role is to store the storage key encountered during a run to store them for debugging purpose.
+  */
+class TestModeWorldStateProxy(
```

Comment:
Why not a case class?

Reply:
Yeah you're right. I blindly copied the logic from `InMemoryWorldStateProxy`. But a case class will give us the copy for free.